tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/IRChecker.scala

```diff
@@ -1060,9 +1060,16 @@ class IRChecker(unit: LinkingUnit, logger: Logger) {
 }
 
 object IRChecker {
-  private final class ErrorContext private (pos: Position, name: String) {
-    override def toString(): String =
+  private final class ErrorContext private (val treeOrLinkedClass: Any)
```

Comment:
I was indeed concerned about allocations. Subclasses don't solve the other part of the problem, which is the allocation of the `ErrorContext` itself. It can't be an `AnyVal` if there are two subclasses.